bridge/tfchain_bridge/pkg/bridge/withdraw.go

```diff
@@ -41,6 +42,7 @@ func (bridge *Bridge) handleWithdrawCreated(ctx context.Context, withdraw subpkg
 
 	// check if it can hold tft : TODO check trust line TFT limit if it can receive the amount
 	if err := bridge.wallet.CheckAccount(withdraw.Target); err != nil {
+		ctx = _logger.WithRefundReason(ctx, err.Error())
 		return bridge.handleBadWithdraw(ctx, withdraw)
```

Comment:
So far, so good. I'm just considering how we handle failures of any kind and the process of issuing refunds.

In the event that the Horizon server goes down and all retry attempts are unsuccessful, a refund will be issued on TFchain. In this case, the user will cover the withdrawal(burn) fees and deposit(mint) fees and take responsibility for the outage.

We could take a more user-friendly approach. Instead, if the issue is a server error, we can choose to simply skip(postpone) the withdraw, as we can rely on TFchain to notify us again. For client-side errors (such as account not found or TFT trust line not added), we should issue a refund.

Let me know what do you think.